packages/pigeon/lib/objc_generator.dart

```diff
@@ -0,0 +1,217 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'ast.dart';
+import 'generator_tools.dart';
+
+/// Options that control how Objective-C code will be generated.
+class ObjcOptions {
+  /// Parametric constructor for ObjcOptions.
+  ObjcOptions({this.header, this.prefix});
+
+  /// The path to the header that will get placed in the source filed (example:
+  /// "foo.h").
+  String header;
+
+  /// Prefix that will be appended before all generated classes and protocols.
+  String prefix;
+}
+
+String _className(String prefix, String className) {
+  if (prefix != null) {
+    return '$prefix$className';
+  } else {
+    return className;
+  }
+}
+
+const Map<String, String> _objcTypeForDartTypeMap = <String, String>{
```

Comment:
Does this handle maps or generic list types?

Reply:
No, i worked off the documentation for platform channels.  If StandardMessageCodec supports it I can add it later.

Reply:
It does support it yeah. It's not super clear but that's what it means by `List` and `Map` in https://flutter.dev/docs/development/platform-integration/platform-channels#codec.

If you pass a `List<Map<String, List<int>>` across the platform channels it will involve some boilerplate to serialize and unparse at each end, but the channel does support it. That specific type is convoluted but as a more realistic example we have a bunch of plugins that pass across effectively a `Map<String, dynamic>` that then cast that `dynamic` to stuff like `List<int>` or even `List<Map<String, dynamic>>` later on. The in_app_purchase plugin is probably the biggest example.

Reply:
Added note to README about generics not being supported yet.